File: backend/internal/domain/entity/space.go
package entity

import "time"

type SpaceVisibility string

const (
	VisibilityPrivate SpaceVisibility = "private"
	VisibilityPublic  SpaceVisibility = "public"
)

type Space struct {
	ID          uint64
	Name        string
	Description string
	Icon        string
	Cover       string
	Visibility  SpaceVisibility
	OwnerID     uint64
	CreatedAt   time.Time
	UpdatedAt   time.Time
}

File: frontend/src/types/space.ts
export interface Space {
  id: number
  name: string
  description: string
  icon: string
  cover: string
  visibility: 'private' | 'public'
  owner_id: number
  created_at: string
  updated_at: string
}

export interface SpaceMember {
  id: number
  space_id: number
  user_id: number
  role: 'admin' | 'member' | 'viewer'
  created_at: string
}

export interface CreateSpaceRequest {
  name: string
  description?: string
  icon?: string
  cover?: string
  visibility?: string
}
